### ISSUE TITLE
Multi-word search terms not working in Track Requests

### ISSUE THREAD

**Issue by candu** [CityofToronto/bdit_flashcrow#871]
**What Happened**
From [user testing spreadsheet (internal only)](https://docs.google.com/spreadsheets/d/1ixtkOYs1RLO1DKO-7-_6MpnfNhEvSgRt70JUIstI6N0/edit#gid=1768524490):

> Only works when typing in the first letter of "Turning Movement Count" and "Multiple Locations", and the word "Ped" of "Ped Delay and Classification", and the word "Speed" of "Speed / Volume ATR". Does not work when full word is typed out, and if there are spaces between words.

**What Should Happen**
This is...a good question!  There are a few possible resolutions:

- decide that the current behaviour is OK, and leave it;
- remove some of the search columns from the list (i.e. Type, Assigned, Status), and expect users to use filters instead;
- fix these search columns (see Additional Notes below).

**To Reproduce**
Steps to reproduce the bug:
1. Log in as any user with `STUDY_REQUESTS` permissions;
2. Go to "Track Requests" using the left navbar;
3. Enter "Turning" in the search bar.

At this point, you'd expect Turning Movement Counts (TMCs) to come up in search results, but they don't.

**Additional Notes**
The underlying issue is that, in database, we use enum codes from `StudyType`, `StudyRequestAssignee`, and `StudyRequestStatus` - not the full text.  Moreover, these enum codes aren't kept in the database, but are instead listed in code (see `@/lib/Constants`):

```js
StudyType.init({
  // other keys...
  TMC: {
    label: 'Turning Movement Count',
    // other properties...
  },
});
```

Here the `label` is what shows up in the frontend, but it's stored in the database as `TMC`.

On top of this: to facilitate searching, we've introduced the table `study_request_items`, which combines non-bulk requests with aggregated values across bulk requests to form an index structure to support filter, search, and sort.  This is what allows us to treat bulk and non-bulk requests identically in the table.

This means that, while we *could* just store the `label` values in `study_request_items`...what if they change?  (For instance: we're already talking about re-labelling `ATR_VOLUME_BICYCLE`.)  That would mean rebuilding all of `study_request_items`, adding more maintenance behind the scenes (and more potential for surprises if that maintenance doesn't happen!)

Personally, I'd recommend we *don't* offer search on status, assignee, or type: these are much, much easier to handle as filters.  (That would leave ID, location, and requester as the only searchable columns.)  @shinechaudhuri @mkewins - thoughts?

**Comment by mkewins** [comment]
tl;dr: I am leaning towards option 2, pull search from type, assigned, status.

More on my thinking why below.

Search makes sense on things that are not a fixed set -- id, location, requester -- since we won't ever add these to a list of filters. You might be looking for a specific request id, requests from a specific person, or for a specific location. And we have no way of knowing the full set of options, since new staff might join over time, who knows how high the number of id's could get! Search would help narrow that down.

For properties where the options are from a fixed set (i.e. study type, assignee, status), those probably make the most sense to filter down by. Especially since you might want to use two, or more, of these at once. For example, find TMCs (study type) assigned (status) to the contractor (assignee) but you wouldn't use search for all that.
